### PR TITLE
MainService: crop image if padding is present

### PR DESCRIPTION
**Pull request by eisaev**
On some devices we have some padding on the right side of the image, like this:
Screenshot 2023-03-11 at 16 40 10
This change will crop the image if necessary. Result:
Screenshot 2023-03-11 at 16 38 59

### CODE REVIEW

**Comment by bk138** [comment]
Thanks for the contributions and sorry for the delay! 

Some thoughts:
* This fixes an aesthetic issue at the cost of two framebuffer copies.
* As such, I'm a little bit concerned that user that prefer perfomance over beauty will complain ;-)
* A way out would be to have this configurable via the Admin UI, feel free to update your PR accordingly.

**Comment by eisaev** [comment]
@bk138 
> A way out would be to have this configurable via the Admin UI, feel free to update your PR accordingly.

I added `Remove padding` option to the Admin UI with default value `false`. Thanks for the suggestion!

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <string name="main_activity_settings_password">Password</string>
     <string name="main_activity_settings_start_on_boot">Start on Boot</string>
     <string name="main_activity_settings_scaling">Scaling</string>
+    <string name="main_activity_settings_remove_padding">Remove padding</string>
```
Comment (bk138) [inline app/src/main/res/values/strings.xml:9]: please make it "Remove Padding" (in title case). As an extra bonus, you could add this to the other translation as well using Google Translate :heart: .

File: app/src/main/java/net/christianbeier/droidvnc_ng/Defaults.kt [hunk 2]
```diff
@@ -68,6 +72,7 @@ class Defaults {
             this.portRepeater = readDefault.portRepeater
             this.scaling = readDefault.scaling
             this.password = readDefault.password
+            this.removePadding = readDefault.removePadding
```
Comment (bk138) [inline app/src/main/java/net/christianbeier/droidvnc_ng/Defaults.kt:75]: If you could also update the example JSON in the README, that'd be awesome :-)

**Review by bk138 — CHANGES_REQUESTED** [review]
Thanks for this well-made PR! Only some minor requests!

File: app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java [hunk 3]
```diff
@@ -23,10 +23,11 @@
 
 public class Constants {
     public static final String PREFS_KEY_SETTINGS_PORT = "settings_port";
-    public static final String PREFS_KEY_SETTINGS_PASSWORD = "settings_password" ;
-    public static final String PREFS_KEY_SETTINGS_START_ON_BOOT = "settings_start_on_boot" ;
-    public static final String PREFS_KEY_SETTINGS_SCALING = "settings_scaling" ;
-    public static final String PREFS_KEY_REVERSE_VNC_LAST_HOST = "reverse_vnc_last_host" ;
-    public static final String PREFS_KEY_REPEATER_VNC_LAST_HOST = "repeater_vnc_last_host" ;
-    public static final String PREFS_KEY_REPEATER_VNC_LAST_ID = "repeater_vnc_last_id" ;
+    public static final String PREFS_KEY_SETTINGS_PASSWORD = "settings_password";
```
[the remark this answers is not in the document]
Comment (eisaev) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java:26]: There were spaces before the semicolons. Are you sure that they should be there?

[the remark this answers is not in the document]
Comment (bk138) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java:26]: You're right, they should not be there. But I'm a super big fan of PRs that change _one_ thing. But let's not be to picky and leave your rightful changes in there.

[the remark this answers is not in the document]
Comment (eisaev) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java:26]: I also prefer to do only one thing per PR. But in this case there are no unrelated changes in logic. These are just white spaces. Anyway, I appreciate your opinion and will remove these changes from PR.

[the remark this answers is not in the document]
Comment (bk138) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java:26]: As you like!

[the remark this answers is not in the document]
Comment (eisaev) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Constants.java]: Reverted in ba78bb6

File: app/src/main/java/net/christianbeier/droidvnc_ng/MainActivity.java [hunk 4]
```diff
@@ -287,6 +287,14 @@ public void afterTextChanged(Editable editable) {
             ed.apply();
         });
 
+        final SwitchMaterial removePadding = findViewById(R.id.settings_remove_padding);
+        removePadding.setChecked(prefs.getBoolean(Constants.PREFS_KEY_SETTINGS_REMOVE_PADDING, mDefaults.getRemovePadding()));
+        removePadding.setOnCheckedChangeListener((compoundButton, b) -> {
+            SharedPreferences.Editor ed = prefs.edit();
+            ed.putBoolean(Constants.PREFS_KEY_SETTINGS_REMOVE_PADDING, b);
+            ed.commit();
```
[the remark this answers is not in the document]
Comment (eisaev) [reply app/src/main/java/net/christianbeier/droidvnc_ng/MainActivity.java]: Fixed in ba78bb6

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (eisaev) [reply app/src/main/res/values/strings.xml:9]: Added in ba78bb6

File: app/src/main/java/net/christianbeier/droidvnc_ng/Defaults.kt (hunk 2 above)
Comment (eisaev) [reply app/src/main/java/net/christianbeier/droidvnc_ng/Defaults.kt:75]: Added in ba78bb6

**Pull request merged**